modules/customizing-the-web-console-URL.adoc

```diff
@@ -67,3 +72,75 @@ spec:
     secret:
       name: console-tls
 ----
+
+[id="customizing-the-download-route_{context}"]
+== Customizing the `download` route
+
+.Procedure
+
+. Modify the cluster instance created by default during installation in the `route.openshift.io` custom resource:
```

Comment:
`ingress` config needs to be used for customizing both routes via the new API. It can' set the host directly on the host since CVO will stomp it to the default value.
The point of the work is that when the admin sets the desired hostname for either `console` or `downloads` route(component), he needs to do it via `ingress` config. After that's being set `console-operator` will pick up those changes on the `ingress` config and reflect them by creating new, custom route for that given component. Once the custom route is create, we are also preserving default routes, mainly for two reasons:
- so we dont break any bookmarks
- so nobody can take the default route hostname

Reply:
To customize the `console` route, admin needs to the set the custom hostname and TLS(optional), on the `ingress` config's spec.

```yaml
spec:
  componentRoutes:
    - name: console
       namespace: openshift-console
       hostname: <custom-hostname>
       secret:
         name: <secret-name>
```

For `downloads` route the the spec will look similar, only the name will change

```yaml
spec:
  componentRoutes:
    - name: downloads
       namespace: openshift-console
       hostname: <custom-hostname>
       secret:
         name: <secret-name>
```

For the console we are checking if the custom hostname has cluster domain suffix, which indicates
if a secret that contains TLS certificate and key needs to exist in the `openshift-config` namespace and referenced in  the operator config. If the suffix matches the cluster domain, then the secret is optional. If the suffix doesn't matches the cluster domain, then the secret is mandatory. Secret has to contain `tls.crt` and `tls.key` data keys where the certificate and key are stored and both need to be in valid format.